src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -698,6 +708,47 @@ def set_up_system_assigned_identity_as_default_if_using_acr(self):
                         return
                 self.set_argument_registry_identity('system')
 
+    def set_up_consumption_gpu_wp_payload(self, consumption_gpu_profile_type):
+        consumption_gpu_profile_type_lower = consumption_gpu_profile_type.lower()
+        if consumption_gpu_profile_type_lower == "consumption-gpu-nc8as-t4":
+            payload = {
+                "workloadProfileType": "Consumption-GPU-NC8as-T4",
+                "name": "consumption-8core-t4"
+            }
+        elif consumption_gpu_profile_type_lower == "consumption-gpu-nc4as-t4":
+            payload = {
+                "workloadProfileType": "Consumption-GPU-NC4as-T4",
+                "name": "consumption-4core-t4"
+            }
+        elif consumption_gpu_profile_type_lower == "consumption-gpu-nc24-a100":
+            payload = {
+                "workloadProfileType": "Consumption-GPU-NC24-A100",
+                "name": "consumption-24core-a100"
+            }
+        elif consumption_gpu_profile_type_lower == "consumption-gpu-nc12-a100":
+            payload = {
+                "workloadProfileType": "Consumption-GPU-NC12-A100",
+                "name": "consumption-12core-a100"
+            }
```

Comment:
Currently when we want to support a new type of workload profile, we don't need to update CLI.
These hard codes are coupled to the workload profile type, which is not conducive to future expansion.

Reply:
Hi @harryli0108  @Tratcher 

Do we really need add functionality to create gpu app for `az containerapp create`?
Currently we use `az containerapp create  --workload-profile-name` to specify the workload profile to run the app on.
This --consumption-gpu-profile has conflict with it.

Reply:
@harryli0108 
I think it would be better to use `az containerapp up` for adding functionality to create gpu app.
We can prepare the managed environment workload profile ion `az containerapp up` prepare logic.
```
Command
    az containerapp up : Create or update a container app as well as any associated resources (ACR,
    resource group, container apps environment, GitHub Actions, etc.).
```